### PR TITLE
Do not duplicate phpcr-transport value in test workflow

### PR DESCRIPTION
**Pull request by niklasnatter**

### CODE REVIEW

File: .github/workflows/test-application.yaml [hunk 1]
```diff
@@ -34,11 +34,10 @@ jobs:
...
 
     test:
-        name: "PHP ${{ matrix.php-version }} (${{ matrix.database }}, ${{ matrix.phpcr-transport }}, ${{ matrix.dependency-versions }})"
+        name: "PHP ${{ matrix.php-version }} (${{ matrix.database }}, ${{ matrix.env.SYMFONY__PHPCR__TRANSPORT }}, ${{ matrix.dependency-versions }})"
         runs-on: ubuntu-latest
 
         env:
-            JACKRABBIT_VERSION: 2.20.2
             SYMFONY_DEPRECATIONS_HELPER: weak
             SYMFONY__DATABASE__HOST: 127.0.0.1
```
Comment (luca-rath) [inline .github/workflows/test-application.yaml]: Would prefer to do `SYMFONY__PHPCR__TRANSPORT: ${{ matrix.phpcr-transport }}` here

File: .github/workflows/test-application.yaml [hunk 2]
```diff
@@ -121,19 +118,19 @@ jobs:
                   coverage: none
 
             - name: Download jackrabbit
-              if: ${{ matrix.phpcr-transport == 'jackrabbit' }}
+              if: ${{ matrix.env.SYMFONY__PHPCR__TRANSPORT == 'jackrabbit' }}
               uses: carlosperate/download-file-action@v1.0.3
               with:
-                  file-url: http://archive.apache.org/dist/jackrabbit/${{ env.JACKRABBIT_VERSION }}/jackrabbit-standalone-${{ env.JACKRABBIT_VERSION }}.jar
+                  file-url: http://archive.apache.org/dist/jackrabbit/${{ matrix.env.JACKRABBIT_VERSION }}/jackrabbit-standalone-${{ matrix.env.JACKRABBIT_VERSION }}.jar
```
Comment (luca-rath) [inline .github/workflows/test-application.yaml]: If you add `env: ${{ matrix.env }}` at the end of this step, you can just use `${{ env.JACKRABBIT_VERSION }}`

File: .github/workflows/test-application.yaml [hunk 3]
```diff
@@ -155,29 +152,29 @@ jobs:
...
         env:
-            JACKRABBIT_VERSION: 2.20.2
             SYMFONY_DEPRECATIONS_HELPER: weak
-            SYMFONY__PHPCR__TRANSPORT: jackrabbit
-            SYMFONY__DATABASE__DRIVER: pdo_mysql
-            SYMFONY__DATABASE__VERSION: 5.7
             SYMFONY__DATABASE__HOST: 127.0.0.1
-            SYMFONY__DATABASE__PORT: 3306
-            SYMFONY__DATABASE__USER: root
-            SYMFONY__DATABASE__PASSWORD: root
```
Comment (luca-rath) [inline .github/workflows/test-application.yaml]: keeping `SYMFONY_DEPRECATIONS_HELPER` here is okay to set this globally, but why do you keep `SYMFONY__DATABASE__HOST` here? This shouldn't be treated differently from env vars like `SYMFONY__DATABASE__PORT`

**Review by luca-rath — CHANGES_REQUESTED** [review]

File: .github/workflows/test-application.yaml (hunk 1 above)
Comment (luca-rath) [reply .github/workflows/test-application.yaml]: Then you can continue using `${{ matrix.phpcr-transport }}` which is less verbose

Comment (niklasnatter) [reply .github/workflows/test-application.yaml]: sounds good 🙂

File: .github/workflows/test-application.yaml (hunk 3 above)
Comment (niklasnatter) [reply .github/workflows/test-application.yaml]: just copied the section from above to be honest 😅  will move the `SYMFONY__DATABASE__HOST`

File: .github/workflows/test-application.yaml [hunk 4]
```diff
@@ -55,9 +53,10 @@ jobs:
                       tools: 'composer:v1'
                       php-cs-fixer: false
                       env:
-                          SYMFONY__PHPCR__TRANSPORT: doctrinedbal
+                          SYMFONY__PHPCR__TRANSPORT: ${{ matrix.phpcr-transport }}
```
Comment (luca-rath) [inline .github/workflows/test-application.yaml]: Maybe this works, if you do that in the `env` of the job and not here in the matrix definition

File: .github/workflows/test-application.yaml [hunk 5]
```diff
@@ -38,9 +38,7 @@ jobs:
         runs-on: ubuntu-latest
 
         env:
-            JACKRABBIT_VERSION: 2.20.2
             SYMFONY_DEPRECATIONS_HELPER: weak
```
Comment (luca-rath) [inline .github/workflows/test-application.yaml:41]:
```suggestion
            SYMFONY_DEPRECATIONS_HELPER: weak
            SYMFONY__PHPCR__TRANSPORT: ${{ matrix.phpcr-transport }}
```

File: .github/workflows/test-application.yaml (hunk 4 above)
Comment (luca-rath) [inline .github/workflows/test-application.yaml]:
```suggestion
```

File: .github/workflows/test-application.yaml [hunk 6]
```diff
@@ -69,9 +68,10 @@ jobs:
                       tools: 'composer:v2'
                       php-cs-fixer: false
                       env:
-                          SYMFONY__PHPCR__TRANSPORT: doctrinedbal
+                          SYMFONY__PHPCR__TRANSPORT: ${{ matrix.phpcr-transport }}
```
Comment (luca-rath) [inline .github/workflows/test-application.yaml]:
```suggestion
```

File: .github/workflows/test-application.yaml [hunk 7]
```diff
@@ -83,12 +83,14 @@ jobs:
                       tools: 'composer:v2'
                       php-cs-fixer: true
                       env:
-                          SYMFONY__PHPCR__TRANSPORT: jackrabbit
+                          SYMFONY__PHPCR__TRANSPORT: ${{ matrix.phpcr-transport }}
```
Comment (luca-rath) [inline .github/workflows/test-application.yaml]:
```suggestion
```

File: .github/workflows/test-application.yaml [hunk 8]
```diff
@@ -159,15 +162,7 @@ jobs:
         runs-on: windows-latest
 
         env:
-            JACKRABBIT_VERSION: 2.20.2
             SYMFONY_DEPRECATIONS_HELPER: weak
```
Comment (luca-rath) [inline .github/workflows/test-application.yaml:162]:
```suggestion
            SYMFONY_DEPRECATIONS_HELPER: weak
            SYMFONY__PHPCR__TRANSPORT: ${{ matrix.phpcr-transport }}
```

File: .github/workflows/test-application.yaml [hunk 9]
```diff
@@ -178,6 +173,15 @@ jobs:
                       dependency-versions: 'highest'
                       php-extensions: 'ctype, iconv, mysql, pdo_mysql, php_fileinfo, imagick'
                       tools: 'composer'
+                      env:
+                          SYMFONY__PHPCR__TRANSPORT: ${{ matrix.phpcr-transport }}
```
Comment (luca-rath) [inline .github/workflows/test-application.yaml]:
```suggestion
```

**Review by luca-rath — APPROVED** [review]

**Review by alexander-schranz — APPROVED** [review]

**Review by luca-rath — APPROVED** [review]

**Pull request merged**